### ISSUE TITLE
Make `.info_complete` a property?

### ISSUE THREAD

**Issue by dstansby** [zarr-developers/zarr-python#2600]
I noticded that there's a new [`info_complete` method](https://zarr--2589.org.readthedocs.build/en/2589/_autoapi/zarr/index.html#zarr.Array.info_complete), which looks super neat! I find it a bit confusing that `.info `is a property, but `.info_complete`is a method though.

Would it be worth changing `info_complete` to a property to match? Or `info` to a method to match `info_complete`?

**Comment by jhamman** [comment]
@TomAugspurger will likely be able to give the best reasoning for why we landed on a method. IIRC, it had to do with signaling that `info_complete` would trigger (potentially expensive) IO.

**Comment by TomAugspurger** [comment]
That's correct. info is a property for backwards compatibility. info_complete does I/O, so I wanted it to be awaitable.
